extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/MSQTasks.java

```diff
@@ -155,7 +156,11 @@ static String getHostFromSelfNode(@Nullable final DruidNode selfNode)
   static StorageConnector makeStorageConnector(final Injector injector)
   {
     try {
-      return injector.getInstance(Key.get(StorageConnector.class, MultiStageQuery.class));
+      StorageConnector storageConnector = injector.getInstance(Key.get(StorageConnector.class, MultiStageQuery.class));
+      if (storageConnector instanceof NilStorageConnector) {
+        throw new Exception("Storage connector not configured.");
```

Comment:
That is also one approach. Since we would need to check this in the broker as well, I thought we can make it similar in all places. 

I think what I can do is keep doing this check but also bind the NilStorageConnector only to the broker role.